### PR TITLE
V2 Multi-select (Team Select)

### LINKED ISSUE #3856: 2.0 Single & Mult-Select + Dropdown {Component}
[Select in Figma](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=385%3A33491) | [Select Dropdown in Figma](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=1388%3A67631)

Image

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

Adds V2 design for multi-select

image

Fixes #3856 

[Loom](https://www.loom.com/share/63342a2f92d14e8e89c5dd11a4dbf41c)

**Note:** We would be adding avatars to the selected user chips as a follow up as it requires a bit more time to override the built in chip display and function by react-select. Should be done by V2.2, if I am to guess rn.

**Environment**: Staging(main branch)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Jaibles** [comment]
@sean-brydon has raised the point that the previous way works really well now with auto-animate and I tend to agree with him. We will need this multi-select component for other things, but given the space we have free in this instance - I think we might be better with what we had before.. sorry!

**Comment by alishaz-polymath** [comment]
> @sean-brydon has raised the point that the previous way works really well now with auto-animate and I tend to agree with him. We will need this multi-select component for other things, but given the space we have free in this instance - I think we might be better with what we had before.. sorry @alishaz-polymath, my bad!

That's fine. Yeah, I think we can probably use it elsewhere. I'll just remove the usage of it in the Team Member list and just leave it as a module to be used if and when we come around it. 👍 No worries.

File: packages/ui/v2/modules/event-types/MultiDropdownSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  if (!props.hasValue) {
+    return <components.ValueContainer {...props}>{children as React.ReactNode[]}</components.ValueContainer>;
+  }
+  const CHIPS_LIMIT = 2;
+  // TODO:: fix the following ts error
+  // @ts-expect-error: @see children is an array but identified as object resulting in the error
+  const [chips, other] = children;
+  const overflowCounter = chips.slice(CHIPS_LIMIT).length;
+  const displayChips = chips.slice(overflowCounter, overflowCounter + CHIPS_LIMIT);
+  // console.log(props.getValue(), props.clearValue);
```
Comment (baileypumfleet) [inline packages/ui/v2/modules/event-types/MultiDropdownSelect.tsx:23]: Is there any reason to keep this console.log?

Comment (alishaz-polymath) [reply packages/ui/v2/modules/event-types/MultiDropdownSelect.tsx:23]: Oh, that's a mistake. No, we don't need it. I was debugging the type errors and somehow missed on removing it entirely

**Review by PeerRich — APPROVED** [review]
good stuff. will use later. 

maybe for adding guests on the public booking page?

**Comment by alishaz-polymath** [comment]
> good stuff. will use later.
> 
> maybe for adding guests on the public booking page?

Actually not because options have to be pre defined and in the public page, guest emails aren't predefined. But I'm sure there will be use case of this in the future. I'll work on it a bit more to see if I can ass the avatar to the chips and maybe even animation to see if we have a viable option for the team member select, but that's not a priority at this stage as @Jaibles mentioned.

**Pull request merged**